apps/routes/lib/repo.ex

```diff
@@ -47,11 +47,13 @@ defmodule Routes.Repo do
     end
   end
 
-  @spec get_shapes(String.t(), 0 | 1, boolean) :: [Shape.t()]
-  def get_shapes(route_id, direction_id, filter_negative_priority? \\ true) do
+  @spec get_shapes(String.t(), Keyword.t(), boolean) :: [Shape.t()]
+  def get_shapes(route_id, opts, filter_negative_priority? \\ true) do
+    opts = Keyword.put(opts, :route, route_id)
+
     shapes =
-      cache({route_id, direction_id}, fn _ ->
-        case Shapes.all(route: route_id, direction_id: direction_id) do
+      cache(opts, fn _ ->
```

Comment:
are the same `opts` always going to be in the same order? If not, we might be hurting our cache hit rate.

Reply:
Easy to fix by changing `opts` to `Enum.sort(opts)`